leasing/management/commands/import_periodic_rent_adjustment_index.py

```diff
@@ -0,0 +1,407 @@
+import datetime
+import json
+from typing import Callable, TypedDict
+
+import requests
+from django.core.management.base import BaseCommand, CommandError
+from django.utils.dateparse import parse_datetime
+
+from leasing.models.rent import IndexNumber, OldDwellingsInHousingCompaniesPriceIndex
+
+
+class IndexInput(TypedDict):
+    """The details we need to import an index from Tilastokeskus API."""
+
+    name: str
+    url: str
+    code: str
+
+
+class ColumnItem(TypedDict, total=False):
+    """Details of a column in API response data.
+
+    Based on section 4.5 in API docs https://pxdata.stat.fi/API-description_SCB.pdf
+
+    Keys:
+        code: Identifier for the column.
+        text: Textual display name for the column.
+        type: Type of the column: "d" = dimension, "t" = time, "c" = measure. \
+              Optional. Default: "d".
+        unit: Unit for the measures. Only for type "c". Optional.
+        comment: Optional.
+    """
+
+    code: str
+    text: str
+    type: str
+    unit: str
+    comment: str
+
+
+class CommentItem(TypedDict):
+    """A single comment item for a value in API response data.
+
+    Keys:
+        variable: Identifies the column.
+        value: Identifies the value.
+        comment: A comment for the specified value.
+    """
+
+    variable: str
+    value: str
+    comment: str
+
+
+class DataPoint(TypedDict):
+    """A single data point from the API response data.
+
+    All keys and values should be in same order as "columns" items in API
+    response data.
+
+    Keys:
+        key: List of all dimension and time column values.
+        values: List of all measure column values.
+    """
+
+    key: list[str]
+    values: list[str]
+
+
+class MetadataItem(TypedDict):
+    """Metadata details for the database table that the response is based on."""
+
+    updated: str
+    label: str
+    source: str
+
+
+class ResponseData(TypedDict):
+    """The entire data part of the API response."""
+
+    columns: list[ColumnItem]
+    comments: list[CommentItem]
+    data: list[DataPoint]
+    metadata: list[MetadataItem]
+
+
+class ResponseDataError(Exception):
+    """When the data from API response might cause the program to malfunction."""
+
+    pass
+
+
+INDEXES_TO_IMPORT: list[IndexInput] = [
+    {
+        "name": "13mp -- Price index of old dwellings in housing companies (2020=100) \
+                and numbers of transactions, quarterly",
+        "url": "https://pxdata.stat.fi:443/PxWeb/api/v1/en/StatFin/ashi/statfin_ashi_pxt_13mp.px",
+        "code": "ketj_P_QA_T",
+    },
+]
+
+
+class Command(BaseCommand):
+    help = "Import the indexes related to Periodic Rent Adjustment (Tasotarkistus) \
+            from Tilastokeskus database."
+
+    def handle(self, *args, **options):
+        for index_input in INDEXES_TO_IMPORT:
+            self.stdout.write(f'Index: "{index_input["name"]}"')
+
+            index_data = _get_index_data(index_input["url"], index_input["code"])
```

Comment:
This one accesses network, so unit testing as such is not stable. Maybe we could create a dummy data set that is representative of this function's return value, to use it to test other functions in this function.